### PR TITLE
feat: make default message widget easier to reuse

### PR DESCRIPTION
**Pull request by imtoori**

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 1]
```diff
@@ -56,32 +59,37 @@ typedef ReplyTapCallback = void Function(Message);
...
-  bool? isLastUser;
+  late final bool isLastUser;
 
   /// True if the user message is the same of the next message
-  bool? isNextUser;
+  late final bool isNextUser;
 
   /// The message
-  Message message;
+  late final Message message;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:89]: Not really needed
```suggestion
  final Message message;
```

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 2]
```diff
@@ -56,32 +59,37 @@ typedef ReplyTapCallback = void Function(Message);
...
 
   /// The index of the message
-  int index;
+  late final int index;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:92]: Not really needed
```suggestion
  final int index;
```

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 3]
```diff
@@ -1130,6 +1074,21 @@ class _MessageListViewState extends State<MessageListView> {
...
+        MessageDetails(
+          userId,
+          message,
+          messages,
+          index,
+          messageWidget as MessageWidget,
+        ),
+        messages,
+      );
+    }
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:1089]: # Do not commit this
maybe something like this
```suggestion
    if (widget.messageBuilder != null) {
      messageWidget = widget.messageBuilder!(
        context,
        MessageDetails(
          userId,
          message,
          index,
        ),
        messages,
        messageWidget as MessageWidget,
      );
    }
```

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 4]
```diff
@@ -1130,6 +1072,21 @@ class _MessageListViewState extends State<MessageListView> {
...
+    if (widget.messageBuilder != null) {
+      messageWidget = widget.messageBuilder!(
+        context,
+        MessageDetails(
+          userId,
+          message,
+          messages,
+          index,
+        ),
+        messages,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:1084]: two `messages` 😩

Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_list_view.dart:1084]: yea but one is to populate the MessageDetails, the other one it's passed in the builder

Comment (xsahil03x) [reply packages/stream_chat_flutter/lib/src/message_list_view.dart:1084]: but not really needed as part of message details

Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/message_list_view.dart:1084]: but we don't have any instance of the message list in the message details
it's just for computing the message details properties

Comment (xsahil03x) [reply packages/stream_chat_flutter/lib/src/message_list_view.dart:1084]: got it

**Review by deven98 — APPROVED** [review]
LGTM

**Pull request merged**